barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -2628,6 +2628,103 @@ void AvmTraceBuilder::op_call(uint8_t indirect,
     side_effect_counter = static_cast<uint32_t>(hint.end_side_effect_counter);
 }
 
+/**
+ * @brief Static Call with direct or indirect memory access.
+ *
+ * NOTE: we do not constrain this here as it's behaviour will change fully once we have a full enqueued function
+ * call in one vm circuit
+ * @param indirect byte encoding information about indirect/direct memory access.
+ * @param gas_offset An index in memory pointing to the first of the gas value tuple (l2Gas, daGas)
+ * @param addr_offset An index in memory pointing to the target contract address
+ * @param args_offset An index in memory pointing to the first value of the input array for the external call
+ * @param args_size The number of values in the input array for the static call
+ * @param ret_offset An index in memory pointing to where the first value of the static call return value should
+ * be stored.
+ * @param ret_size The number of values in the return array
+ * @param success_offset An index in memory pointing to where the success flag (U8) of the static call should be
+ * stored
+ * @param function_selector_offset An index in memory pointing to the function selector of the external call (TEMP)
+ */
+void AvmTraceBuilder::op_static_call(uint8_t indirect,
+                                     uint32_t gas_offset,
+                                     uint32_t addr_offset,
+                                     uint32_t args_offset,
+                                     uint32_t args_size_offset,
+                                     uint32_t ret_offset,
+                                     uint32_t ret_size,
+                                     uint32_t success_offset,
+                                     [[maybe_unused]] uint32_t function_selector_offset)
+{
```

Comment:
This looks fine, but we probably could have it just call some helper that is shared by `op_call` and `op_static_call` since they do almost exactly the same thing?

Also we'll need to track `is_static` (should be in kernel inputs for now, but will be in the current "call context" when we support an entire enqueued call). And all the opcodes that create new things (note hashes, nullifiers, new messages, storage writes, ulogs) need to raise the error flag if they are called when `is_static` is high.

Reply:
@dbanks12 Helper implemented and for the side effects issue #8639 was created and assigned to bonobos team